bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitOHItemProxy.java

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.io.homekit.internal;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.common.ThreadPoolManager;
+import org.eclipse.smarthome.core.items.Item;
+import org.eclipse.smarthome.core.library.items.ColorItem;
+import org.eclipse.smarthome.core.library.items.DimmerItem;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.HSBType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * Proxy class that can collect multiple commands for the same openHAB item and merge them to one command.
+ * e.g. Hue and Saturation update for Color Item
+ * 
+ * @author Eugen Freiter Initial contribution
+ *
+ */
+
+public class HomekitOHItemProxy {
+    private static final Logger logger = LoggerFactory.getLogger(HomekitOHItemProxy.class);
+    public static final String HUE_COMMAND = "hue";
+    public static final String SATURATION_COMMAND = "saturation";
+    public static final String BRIGHTNESS_COMMAND = "brightness";
+    public static final String ON_COMMAND = "on";
+
+    // delay, how long wait for further commands.
+    // TODO: make it configurable ?
+    private final int DELAY = 50;
+
+    private final ScheduledExecutorService scheduler = ThreadPoolManager
+            .getScheduledPool(ThreadPoolManager.THREAD_POOL_NAME_COMMON);
+    private ScheduledFuture<?> future;
+
+    private final Item item;
+
+    @Nullable
+    private ConcurrentHashMap<String, State> commandCache;
+
+    public HomekitOHItemProxy(final Item item) {
+        this.item = item;
+    }
+
+    public Item getItem() {
+        return item;
+    }
+
+    private void sendCommand() {
+        if (item instanceof ColorItem) {
+            final State currentState = item.getState() instanceof UnDefType ? HSBType.BLACK : item.getState();
+            final State hue = commandCache.containsKey(HUE_COMMAND) ? commandCache.remove(HUE_COMMAND)
+                    : ((HSBType) currentState).getHue();
```

Comment:
```suggestion
            final State hue = commandCache.getOrDefault(HUE_COMMAND, ((HSBType) currentState).getHue());
```

and clear the total cache at the end of the method.

Reply:
yes, probably better readable.